fluidanimate/src-rl/cellpool.cpp

```diff
@@ -0,0 +1,317 @@
+// Written by Christian Bienia
+// The code in this file implements a memory pool of Cell structures.
+// It serves three purposes:
+//   1.) To minimize calls to malloc and free as much as possible
+//   2.) To reuse cell structures as much as possible
+//   3.) To eliminate unnecessary synchronization for memory allocation
+
+#include <iostream>
+
+#include <stdlib.h>
```

Comment:
to future proof these, I'd change to header files to corresponding c++ ones, e.g., 
```
#include <stdlib.h>
```
is changed to 
```
#include <cstdlib>
```

That will future proof the code a bit as the old header files at some point might not be valid c++ anymore.